homeassistant/components/tessie/binary_sensor.py

```diff
@@ -0,0 +1,144 @@
+    ),
+    TessieBinarySensorEntityDescription(
+        key="charge_state_trip_charging",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TessieBinarySensorEntityDescription(
+        key="climate_state_auto_seat_climate_left",
+        device_class=BinarySensorDeviceClass.HEAT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TessieBinarySensorEntityDescription(
+        key="climate_state_auto_seat_climate_right",
+        device_class=BinarySensorDeviceClass.HEAT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TessieBinarySensorEntityDescription(
+        key="climate_state_auto_steering_wheel_heat",
+        device_class=BinarySensorDeviceClass.HEAT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TessieBinarySensorEntityDescription(
+        key="climate_state_cabin_overheat_protection",
+        device_class=BinarySensorDeviceClass.RUNNING,
+        is_on=lambda x: x == "On",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TessieBinarySensorEntityDescription(
+        key="climate_state_cabin_overheat_protection_actively_cooling",
+        device_class=BinarySensorDeviceClass.HEAT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TessieBinarySensorEntityDescription(
+        key="vehicle_state_dashcam_state",
+        device_class=BinarySensorDeviceClass.RUNNING,
+        is_on=lambda x: x == "Recording",
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TessieBinarySensorEntityDescription(
+        key="vehicle_state_is_user_present",
+        device_class=BinarySensorDeviceClass.PRESENCE,
+    ),
+    TessieBinarySensorEntityDescription(
+        key="vehicle_state_tpms_soft_warning_fl",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TessieBinarySensorEntityDescription(
+        key="vehicle_state_tpms_soft_warning_fr",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TessieBinarySensorEntityDescription(
+        key="vehicle_state_tpms_soft_warning_rl",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    TessieBinarySensorEntityDescription(
+        key="vehicle_state_tpms_soft_warning_rr",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Tessie binary sensor platform from a config entry."""
+    coordinators = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            TessieBinarySensorEntity(coordinator, description)
+            for coordinator in coordinators
+            for description in DESCRIPTIONS
+            if description.key in coordinator.data
+        ]
```

Comment:
```suggestion
            TessieBinarySensorEntity(coordinator, description)
            for coordinator in coordinators
            for description in DESCRIPTIONS
            if description.key in coordinator.data
```

Can be a generator expression here

Reply:
Must have missed this one. Thanks